### PR TITLE
configuration; add staging config

### PR DESCRIPTION
**Pull request by jedireza**
As part of the In-App Subscription work, I needed a build on my physical device that had a bundle ID that matched what is listed in App Store Connect.

This PR creates a `Sandbox` configuration, which I crated by duplicating the `Debug` configuration. The only changes I made after that was to:
- the bundled IDs (no `-dev` postfix)
- pointed AppClip at the `AppClip/AppClip.entitlements` file for a `Sandbox` build so the bundle IDs and domains match the parent app

### CODE REVIEW

**Comment by darinwf** [comment]
Why did you end up forking it from `Debug` instead of `Release`?

The name `Staging` comes from the Android setup where we use it to refer to the Release build signed with a development certificate so that it can be installed on a device and used to verify the release build before we upload it to the store. I think to minimize confusion we should probably reserve the name "Staging" for a build more like this.

If what you really want is a debug build that has the release bundle IDs, then maybe we should pick a different name for it. Hmm, what to call it??

My recommendation:
1. If you don't care about having debug information, then stick with "Staging" but make it a variant of "Release" with just different signing certificate (like we did manually last night).
2. If you do care about having debug information, then stick with what you have here but come up with a name other than "Staging" for it.

**Comment by jedireza** [comment]
In my case, I am still in active development mode. And I don't want to limit debugging. So option two sounds better. Just need to pick a name for it.

**Comment by jedireza** [comment]
I've renamed this to `Sandbox` since that's the environment I'm testing against in the case of working with App Store Connect purchases/products.

**Review by darinwf — COMMENTED** [review]
That works.

To paint the bikeshed a bit, here's a naming scheme that came to mind:

```
Release
Release (Staging)
Release (Debug)
Dev (Profile)
Dev (Debug)    <--- default configuration
```

Last four configurations would use development certs. `Dev` would be a configuration that uses the `-dev` suffix Build ID but is built with optimizations enabled and debugging disabled. This would be useful for profiling on device. Could also call this one `Dev (Profile)`.

Mentioning this as a comprehensive scheme that we could use.

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8869,7 +8869,7 @@
 					"-framework",
 					SafariServices,
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = "co.neeva.app.ios.browser.SafariExtension";
+				PRODUCT_BUNDLE_IDENTIFIER = co.neeva.app.ios.browser.SafariExtension;
```
[the remark this answers is not in the document]
Comment (jedireza) [reply Client.xcodeproj/project.pbxproj:8872]: Xcode did this itself.

**Comment by darinwf** [comment]
> Mentioning this as a comprehensive scheme that we could use.

btw, no need to define all of these configurations now. would work to call the new one "Release (Debug)" and then we can do the rest later. could be valuable to rename "Debug" to "Dev (Debug)"

**Comment by jedireza** [comment]
Cool. I renamed `Sandbox` to `Release (Debug)` and `Debug` to `Dev (Debug)`.

**Comment by darinwf** [comment]
LGTM, last nit though might be to sort them so that the `Release` configs are next to one another:

Screen Shot 2022-07-13 at 10 17 03 AM

**Review by darinwf — APPROVED** [review]

**Comment by chung-lai** [comment]
> Looks like build is failing. Do we need to specify the build configuration in circleci config? Not sure `build-for-testing` is meant to pick up `Debug` configuration by default.

hmm...but the error says `** BUILD INTERRUPTED **` though, doesn't seem like it's complaining about an actual build failure

**Comment by jedireza** [comment]
Ok after I renamed `Dev (Debug)` back to `Debug` the build passed. I'm going to merge this as-is.

**Pull request merged**